src/commands.cpp

```diff
@@ -1998,173 +1998,281 @@ void CommandUserRequest::procresult()
     client->app->invite_result(e);
 }
 
-CommandPutUA::CommandPutUA(MegaClient* client, const char *an, const byte* av, unsigned avl)
+CommandPutUA::CommandPutUA(MegaClient* client, const char *an, const byte* av, unsigned avl, int ctag)
 {
-    cmd("up");
+    this->an = an;
+    this->av.assign((const char*)av, avl);
 
-    // if removing avatar, do not Base64 encode the attribute value
-    if (!strcmp(an, "+a") && !strcmp((const char *)av, "none"))
-    {
-        arg(an,(const char *)av, avl);
-    }
-    else
+    cmd("upv");
+
+    beginarray(an);
     {
-        arg(an, av, avl);
+        // if removing avatar, do not Base64 encode the attribute value
+        if (!strcmp(an, "+a") && !strcmp((const char *)av, "none"))
+        {
+            element((const char*)av);
+        }
+        else
+        {
+            element(av, avl);
+        }
+
+        const string *attrv = client->ownuser()->getattrversion(an);
+        if (attrv)
+        {
+            element(attrv->c_str());
```

Comment:
The number of elements can be 1 or 2 indeed. If you don't send a version because it's unknown at the moment of the request, then the API may reject it with `API_EEXPIRED` if the attribute already exists and has a version attached. In that case, you need to provide the latest version of the attribute to be able to update it.
This mechanism aims to prevent modification of user attributes that require to be aware of the latest version of the attribute, such as authentication rings or keys.